### ISSUE TITLE
invalidateQueries doesn't work when enabled attribute is set to false

### ISSUE THREAD

**Issue by sayinserdar** [tannerlinsley/react-query#947]
queryCache.invalidateQueries() function doesn't work if the related useQuery definition has { enabled: false } configuration.

Here is the definition of my 'menus' 
![Screen Shot 2020-09-01 at 12 49 46 PM](https://user-images.githubusercontent.com/26919126/91835343-ee230500-ec51-11ea-870e-616fd8c1a772.png)

If I try to create a new 'menu', invalidateQueries works as expected
![Screen Shot 2020-09-01 at 12 51 13 PM](https://user-images.githubusercontent.com/26919126/91835414-06931f80-ec52-11ea-9300-b858990eae24.png)

But if  I add a configuration as { enabled: false} as in the below image invalidateQueries function doesn't work
![Screen Shot 2020-09-01 at 12 50 01 PM](https://user-images.githubusercontent.com/26919126/91835469-190d5900-ec52-11ea-8f75-281b810aa651.png)

I've spent too many hours on this issue. I changed every other details/configuration but nothing helped. I also tried my other queryKeys other than 'menus' but the problem remains. In my case, it's not a big deal to remove the { enable: false } configuration but is this the expected output ? 

 - OS: macOS
 - Browser: Any browser
 - 10.15.3

React Query - v2.12.1
React Query Developer Tools - v2.3.2

Apart from this issue React Query has been smooth and wonderful. Thanks everyone who contributes this project.
Peace

**Comment by boschni** [comment]
What happens if you set `refetchInactive` to `true`? https://react-query.tanstack.com/docs/api#querycacheinvalidatequeries

**Comment by sayinserdar** [comment]
> What happens if you set `refetchInactive` to `true`? https://react-query.tanstack.com/docs/api#querycacheinvalidatequeries

Unfortunately, nothing has changed. Doesn't invalidate

**Comment by jsangaritar** [comment]
Hi, im using `"react-query": "^3.24.3"` and this issue is still happening, i have a query defined with `enable: false` and when i want to invalidate it does not work with that config.

**Comment by TkDodo** [comment]
disabled and inactive are not the same thing. Disabled queries will never be refetched, except with refetch() returned from useQuery. Documented here: https://react-query.tanstack.com/guides/disabling-queries

**Comment by Joshuafolorunsho** [comment]
Thanks for the clarification @TkDodo. I tried passing down refetch so I can use it when I mutate onSuccess. But it is not updating. Any ideas on how I can refetch the data? Thanks

**Comment by Joshuafolorunsho** [comment]
@jsangaritar, were you able to find a way around it?

**Comment by jsangaritar** [comment]
Yeah, thankfully there is only one child component so i passed the refetch
function as a callback prop. Tho, it would be nice to have another prop or
something like `invalidateQueryOnRefetchDisabled` (not good with names).

*Johnathan Angarita Romero*
*Tech Lead & Front-End Developer *
*Development & UX*

**Comment by TkDodo** [comment]
@jsangaritar I'm still wondering what your use-case is where you disable a query but then want to trigger a refetch of that disabled query from another component... it seems like this doesn't work with RQ, but a bit against it. The query will still be disabled after refetch and you won't benefit from smart revalidations ...

**Comment by TkDodo** [comment]
once you set `enabled: false`, you opt out of many things. `enabled: false` basically means that the query is _not_ ready to run. Lost of people use it conditionally, like: `enabled: !!id` to not fetch while the id is not there yet. It will automatically enable itself once `id` is ready.

Hence it is very much on purpose that invalidation skips disabled queries and this will not change.

For your case, you might also want to not set enabled to hardcoded false, but to a state value that changes from false to true once you click that button. See also [lazy queries](https://tanstack.com/query/v4/docs/guides/disabling-queries#lazy-queries)

**Comment by imagine10255** [comment]
In the case of query is enabled: false, but the user needs to log out and clear him, you can use this method

```typescript
const qc = useQueryClient();
qc.resetQueries([ECacheKey.PROFILE_ME]);
```

Hope it helps someone who has the same problem as me

```
"@tanstack/react-query": "^4.28.0",
```

**Comment by Nightvision53** [comment]
I don't think if there is anyone dumb like me encounters this mistake but I leave it for who does.

Apparently wagmi client has the same method `useQueryClient` and if you use something like VSCode auto import with pushing tab, you can mistakenly import Wagmi like 

```typescript
import {  useQueryClient  } from 'wagmi';
```

instead of 

```typescript
import {  useQueryClient  } from 'react-query';
```

**Comment by bellamylobbyist47** [comment]
> once you set `enabled: false`, you opt out of many things. `enabled: false` basically means that the query is _not_ ready to run. Lost of people use it conditionally, like: `enabled: !!id` to not fetch while the id is not there yet. It will automatically enable itself once `id` is ready.
> 
> Hence it is very much on purpose that invalidation skips disabled queries and this will not change.
> 
> For your case, you might also want to not set enabled to hardcoded false, but to a state value that changes from false to true once you click that button. See also [lazy queries](https://tanstack.com/query/v4/docs/guides/disabling-queries#lazy-queries)





https://stackdev.space/questions/refresh-the-screen-when-the-menu-is-clicked
